### PR TITLE
[SYCL][DOC] Document test-related processes

### PR DESCRIPTION
**Pull request by vladimirlaz**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -20,6 +20,11 @@ is strongly encouraged that you submit the patch to https://llvm.org/ directly.
 See [LLVM contribution guidelines](https://llvm.org/docs/Contributing.html)
 for more information.
 
+**NB**: A change in compiler and runtime should be accompanied with
+corresponding test change.
+See [DPC++ test toolchain](sycl/doc/GetStartedGuide.md#test-dpc-toolchain)
+for more information.
```
Comment (bader) [inline CONTRIBUTING.md:26]:
```suggestion
See [Test DPC++ toolchain](sycl/doc/GetStartedGuide.md#test-dpc-toolchain)
section of Get Started guide for more information.
```

File: sycl/doc/GetStartedGuide.md [hunk 2]
```diff
@@ -280,7 +280,50 @@ command:
 
 ### Test DPC++ toolchain
 
-#### Run regression tests
+There are 3 types of tests which are used for DPC++ toolchain validation:
+* DPC++ in-source LIT tests including [check-llvm](../../llvm/test),
```
Comment (bader) [inline sycl/doc/GetStartedGuide.md:284]:
```suggestion
* DPC++ in-tree LIT tests including [check-llvm](../../llvm/test),
```
"in-tree" seems to be more frequently used term in LLVM documentation.

File: sycl/doc/GetStartedGuide.md [hunk 3]
```diff
@@ -280,7 +280,50 @@ command:
...
+[check-clang](../../clang/test), [check-llvm-spirv](../../llvm-spirv/test) and
+[check-sycl](../../sycl/test) targets stored in this repository. These tests
+should be not have hardware or external software dependencies (e.g OpenCL,
```
Comment (bader) [inline sycl/doc/GetStartedGuide.md:287]:
```suggestion
should be not have hardware (e.g. GPU, FPGA, etc.) or external software dependencies (e.g OpenCL,
```

File: sycl/doc/GetStartedGuide.md [hunk 4]
```diff
@@ -280,7 +280,50 @@ command:
...
+Level Zero, CUDA runtimes). All test which do not follow this approach should
```
Comment (bader) [inline sycl/doc/GetStartedGuide.md:288]:
```suggestion
Level Zero, CUDA runtimes). All tests not following this approach should
```

File: sycl/doc/GetStartedGuide.md [hunk 5]
```diff
@@ -280,7 +280,50 @@ command:
...
+be moved to DPC++ end-to-end tests.
```
Comment (bader) [inline sycl/doc/GetStartedGuide.md:289]: or SYCL-CTS?

File: sycl/doc/GetStartedGuide.md [hunk 6]
```diff
@@ -280,7 +280,50 @@ command:
...
+
+* DPC++ end-to-end (E2E) tests which are extension to
+[LLVM\* test suite](https://github.com/intel/llvm-test-suite/tree/intel/SYCL).
+These tests cover full DPC++ stack including all dependencies (e.g. OpenCL,
+Level Zero, CUDA runtimes).
+
+* SYCL-CTS are official [Khronos\* SYCL\* conformance tests](https://github.com/KhronosGroup/SYCL-CTS).
+They verify SYCL 1.2.1 specification compatibility. All implementation details
```
Comment (bader) [inline sycl/doc/GetStartedGuide.md:297]:
```suggestion
They verify SYCL specification compatibility. All implementation details
```
Let's avoid specifying particular version.

File: sycl/doc/GetStartedGuide.md [hunk 7]
```diff
@@ -280,7 +280,50 @@ command:
...
+or extensions are out of scope for the tests.
+
+#### Contribution to DPC++ tests
+Every product change should be accompanied with corresponding test modification
+(adding new test(s), extending or modifying existing test(s)).
+
+The test location is selected depending on test nature:
+
+ - A test which depends only on tools and libraries produced during build of
+the repository can be put to in-source LIT. Also the tests for a feature under
```
Comment (bader) [inline sycl/doc/GetStartedGuide.md:307]:
```suggestion
the repository can be put to in-tree LIT. Also the tests for a feature under
```

File: sycl/doc/GetStartedGuide.md [hunk 8]
```diff
@@ -280,7 +280,50 @@ command:
...
+active development requiring atomic change for tests and product can be put to
+[sycl/test/on-device](../../sycl/test/on-device) temporarily. It is developer
+responsibility to move the tests to DPC++ E2E test suite once feature is
+stabilized.
```
Comment (bader) [inline sycl/doc/GetStartedGuide.md:311]: Does it make sense to rename "on-device" directory to "E2E-unstable" or something like this to reflect the purpose of these tests better?
I also suggest adding a README file to the root of this directory with similar information clarifying the purpose of these tests, expectations we put on developers, etc.

File: sycl/doc/GetStartedGuide.md [hunk 9]
```diff
@@ -280,7 +280,50 @@ command:
...
+
+ - If SYCL specification has changed (SYCL CTS tests conflict with recent
+version of SYCL specification) or change is required in the way the tests are
+built with DPC++ compiler (defined in
+[FindIntel_SYCL](https://github.com/KhronosGroup/SYCL-CTS/blob/SYCL-1.2.1/master/cmake/FindIntel_SYCL.cmake))
+pull request should be created under
+[KhronosGroup/SYCL-CTS](https://github.com/KhronosGroup/SYCL-CTS) with required
+patch.
+
+
```
Comment (bader) [inline sycl/doc/GetStartedGuide.md:325]:
```suggestion

```

File: sycl/doc/GetStartedGuide.md (hunk 8 above)
Comment (vladimirlaz) [reply sycl/doc/GetStartedGuide.md]: I have added README file but would prefer to rename directory as separate PR when there will be agreement on the exact name.

File: sycl/doc/GetStartedGuide.md [hunk 10]
```diff
@@ -280,7 +280,48 @@ command:
...
+Level Zero or CUDA) should be put to DPC++ E2E test suite following
+[CONTRIBUTING](https://github.com/intel/llvm-test-suite/blob/intel/CONTRIBUTING.md).
+
+ - If SYCL specification has changed (SYCL CTS tests conflict with recent
+version of SYCL specification) or change is required in the way the tests are
+built with DPC++ compiler (defined in
+[FindIntel_SYCL](https://github.com/KhronosGroup/SYCL-CTS/blob/SYCL-1.2.1/master/cmake/FindIntel_SYCL.cmake))
+pull request should be created under
+[KhronosGroup/SYCL-CTS](https://github.com/KhronosGroup/SYCL-CTS) with required
+patch.
```
Comment (bader) [inline sycl/doc/GetStartedGuide.md:323]:
```suggestion
patch.

```

**Review by bader — APPROVED** [review]

**Review by bader — DISMISSED** [review]

File: CONTRIBUTING.md [hunk 11]
```diff
@@ -20,6 +20,11 @@ is strongly encouraged that you submit the patch to https://llvm.org/ directly.
 See [LLVM contribution guidelines](https://llvm.org/docs/Contributing.html)
 for more information.
 
+**NB**: A change in compiler and runtime should be accompanied with
+corresponding test change.
```
Comment (romanovvlad) [inline CONTRIBUTING.md:24]:
```suggestion
corresponding test case.
```
?

**Review by romanovvlad — COMMENTED** [review]
LGTM

File: CONTRIBUTING.md (hunk 11 above)
Comment (vladimirlaz) [reply CONTRIBUTING.md:24]: It can be more then one test case or test removal. "Test change" covers all possible modification "test case" don't.

Comment (pvchupin) [inline CONTRIBUTING.md:24]:
```suggestion
corresponding test changes.
```

File: sycl/doc/GetStartedGuide.md [hunk 12]
```diff
@@ -280,7 +280,49 @@ command:
...
+be moved to DPC++ end-to-end or SYCL-CTS tests.
+
+* DPC++ end-to-end (E2E) tests which are extension to
+[LLVM\* test suite](https://github.com/intel/llvm-test-suite/tree/intel/SYCL).
+These tests cover full DPC++ stack including all dependencies (e.g. OpenCL,
+Level Zero, CUDA runtimes).
+
+* SYCL-CTS are official [Khronos\* SYCL\* conformance tests](https://github.com/KhronosGroup/SYCL-CTS).
+They verify SYCL specification compatibility. All implementation details
+or extensions are out of scope for the tests.
```
Comment (pvchupin) [inline sycl/doc/GetStartedGuide.md:298]: This is some duplication of the section below. Can you please merge these?

File: sycl/test/on-device/README.md [hunk 13]
```diff
@@ -0,0 +1,13 @@
+# DPC++ end-to-end tests for features under development
+It is recommended to have in-tree LIT tests independent from HW (e.g. GPU,
```
Comment (pvchupin) [inline sycl/test/on-device/README.md:2]:
```suggestion

We require to have in-tree LIT tests independent from HW (e.g. GPU,
```

**Review by bader — APPROVED** [review]

**Review by pvchupin — APPROVED** [review]

**Comment by vladimirlaz** [comment]
@kbobrovs could you please have a look, approve?

**Comment by pvchupin** [comment]
I think it's blocked by @intel/llvm-reviewers-runtime . I see @romanovvlad LGTMed above so merging...

**Comment by kbobrovs** [comment]
LGTM

**Pull request merged**